packages/remix-testing/create-remix-stub.tsx

```diff
@@ -62,245 +35,25 @@ type RemixStubOptions = {
   initialIndex?: number;
 };
 
-type IndexStubRouteObject = AgnosticIndexRouteObject & {
-  element?: React.ReactNode;
-  children?: undefined;
-};
-
-type NonIndexStubRouteObject = AgnosticNonIndexRouteObject & {
-  element?: React.ReactNode;
-  children?: StubRouteObject[];
-};
-
-// TODO: once Remix is on RR@6.4 we can just use the native type
-type StubRouteObject = IndexStubRouteObject | NonIndexStubRouteObject;
-
-type RemixConfigFuture = Partial<EntryContext["future"]>;
-
-export function createRemixStub(
-  routes: StubRouteObject[],
-  remixConfigFuture?: RemixConfigFuture
-) {
-  // Setup request handler to handle requests to the mock routes
-  let { dataRoutes, queryRoute } = createStaticHandler(routes);
+export function createRemixStub(routes: RouteObject[]) {
   return function RemixStub({
     initialEntries,
-    initialLoaderData = {},
-    initialActionData,
     initialIndex,
+    initialActionData,
+    initialLoaderData,
```

Comment:
Should we just accept `hydrationData` here?

Reply:
i was thinking that too, im indifferent on it unless you have a usecase where youd test "errors" with it

i was also thinking of omitting some things on the route object (id, caseSensitive, etc)

Reply:
actually yeah if we do that we avoid the need to bring in `RouteData`(and get rid of the `UNSAFE` export) and can use `HydrationState` from the router instead